lib/schedule.js

```diff
@@ -336,12 +336,76 @@ function RecurrenceRule(year, month, date, dayOfWeek, hour, minute, second) {
   this.second = (second == null) ? 0 : second;
 }
 
+RecurrenceRule.prototype.checkValidity = function() {
+  function isValidType(num) {
+    if (Array.isArray(num) || (num instanceof Array)) {
+      return num.every(function(e) {
+        return isValidType(e);
+      });
+    }
+    return !(Number.isNaN(Number(num)) && !(num instanceof Range));
+  }
+  if (this.month !== null && (this.month < 0 || this.month > 11 || !isValidType(this.month))) {
+    throw 'Invalid month: ' + this.month;
+  }
+  if (this.dayOfWeek !== null && (this.dayOfWeek < 0 || this.dayOfWeek > 6 || !isValidType(this.dayOfWeek))) {
+    throw 'Invalid dayOfWeek: ' + this.dayOfWeek;
+  }
+  if (this.hour !== null && (this.hour < 0 || this.hour > 23 || !isValidType(this.hour))) {
+    throw 'Invalid hour: ' + this.hour;
+  }
+  if (this.minute !== null && (this.minute < 0 || this.minute > 59 || !isValidType(this.minute))) {
+    throw 'Invalid minute: ' + this.minute;
+  }
+  if (this.second !== null && (this.second < 0 || this.second > 59 || !isValidType(this.second))) {
+    throw 'Invalid second: ' + this.second;
+  }
+  if (this.date !== null) {
+    if(!isValidType(this.date)) {
+      throw 'Invalid date: ' + this.date;
+    }
+    switch (this.month) {
+      case 0:
+      case 2:
+      case 4:
+      case 6:
+      case 7:
+      case 9:
+      case 11:
+        if (this.date < 1 || this. date > 31) {
+          throw 'Invalid date: ' + this.date;
+        }
+        break;
+      case 3:
+      case 5:
+      case 8:
+      case 10:
+        if (this.date < 1 || this. date > 30) {
+          throw 'Invalid date: ' + this.date;
+        }
+        break;
+      case 1:
+        if (this.date < 1 || this. date > 29) {
+          throw 'Invalid date: ' + this.date;
+        }
+        break;
+      default:
+        if (this.date < 1 || this. date > 31) {
+          throw 'Invalid date: ' + this.date;
+        }
+    }
+  }
+  return true;
+};
+
 RecurrenceRule.prototype.nextInvocationDate = function(base) {
   base = (base instanceof Date) ? base : (new Date());
   if (!this.recurs) {
     return null;
   }
 
+  this.checkValidity();
+
```

Comment:
I think that would work as long as RecurrenceRule.prototype.nextInvocationDate is designed to be a private method.  The tests would need to be rewritten since they are currently directly calling that method.  I can do that, as long as there is some benefit gained from moving it there.